### DISCUSSION TITLE
task 08 projection matrix

### DISCUSSION THREAD

**Discussion by zerochaos000** [chulminy/CIVE497-CIVE700#148] [Q&A]
I have tried many ways to construct the projection matrix but none of them will give me correct world point - camera coordinate conversion.... please help. Attached below is my code:

for i=1:nImg
    K{i}=[fMat{i},0,0;0,fMat{i},0;0,0,1];
    RT=[RMat{i},-RMat{i}*CMat{i}'];
    ImgStruct.PMat{i}=K{i}*RT;
end

thanks!!

**Comment by zaalsabb** [comment]
Are `RMat` and `CMat` defined from the camera coordinate system or the world coordinate system?

**Reply by chulminy** [reply]
Where is px and py? which are the half of width and height of the images, respectively.

**Reply by zerochaos000** [reply]
RMat and CMat are taken directly from the ImgStruct from the nvm file. 
I am not sure where should I put in px and py? Are they part of the K matrix? I am confused as according to the definitions given in virtualSFM, "K = [f, 0 0; 0 f 0; 0 0 1]". Thanks

**Reply by zaalsabb** [reply]
If you set K = [f, 0 0; 0 f 0; 0 0 1] as visualSFM does, then the image coordinates will be (0,0) at the center of the image. However, we want the (0,0) to be at the top left of the image. See the definition for the camera matrix (K) here and how px and py are accounted for:
https://github.com/chulminy/CIVE497-CIVE700/blob/master/lecture/clss14_camera_model/camera_model_W21_v4.pdf
